### PR TITLE
Chat 2: Part 6 - conversation minus messages

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/chat/conversation/info-panel/notifications/container.js [hunk 1]
```diff
@@ -1,157 +1,155 @@
...
+          nextProps._storeChannelWide,
+          nextProps._storeDesktop,
+          nextProps._storeMobile,
+          nextProps._storeMuted
+        )
+      }
     },
-  })
+  }),
+  setDisplayName('LifecycleNotifications')
```
Comment (buoyad) [inline shared/chat/conversation/info-panel/notifications/container.js]: 🍳 This should be right after `connect` if the goal is to make it work with storybook

File: shared/chat/conversation/input-area/mention-handler-hoc/index.desktop.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    _triggerDownArrowCounter = () => {
+      this.setState(({downArrowCounter}) => ({downArrowCounter: downArrowCounter + 1}))
+    }
+
+    _triggerPickSelectedCounter = () => {
+      this.setState(({pickSelectedCounter}) => ({pickSelectedCounter: pickSelectedCounter + 1}))
+    }
+
+    onKeyDown = (e: SyntheticKeyboardEvent<>) => {
+      this.props._onKeyDown(e)
```
Comment (buoyad) [inline shared/chat/conversation/input-area/mention-handler-hoc/index.desktop.js]: 🍳 looks like up/down/tab isn't working to change selection

File: shared/chat/conversation/input-area/channel-mention-hud/mention-hud-container.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// @flow
+import {MentionHud} from '.'
+import {connect, type TypedState} from '../../../../util/container'
+import * as Constants from '../../../../constants/chat2'
+
+const mapStateToProps = (state: TypedState, {filter, conversationIDKey}) => {
```
Comment (buoyad) [inline shared/chat/conversation/input-area/channel-mention-hud/mention-hud-container.js]: 🍳 This is getting triggered on ad-hoc convos with a long list of just `#`:

image

File: shared/chat/conversation/input-area/mention-handler-hoc/index.native.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  pickSelectedCounter: number,
+  mentionFilter: string,
+  channelMentionFilter: string,
+  mentionPopupOpen: boolean,
+  channelMentionPopupOpen: boolean,
+  _selection: {selectionStart: number, selectionEnd: number},
+}
+
+const mentionHoc = (InputComponent: React.ComponentType<Props>) => {
+  class MentionHoc extends React.Component<PropsFromContainer, MentionHocState> {
```
Comment (buoyad) [inline shared/chat/conversation/input-area/mention-handler-hoc/index.native.js]: 🍳 The channel mention HUD doesn't seem to work on mobile at all

File: shared/chat/conversation/input-area/normal/index.desktop.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+          />
+          {this.props.emojiPickerOpen && (
+            <EmojiPicker emojiPickerToggle={this.props.emojiPickerToggle} onClick={this._pickerOnClick} />
+          )}
+          {this.props.isEditing && (
+            <Text type="BodySmallPrimaryLink" onClick={this.props.onCancelEditing} style={cancelStyle}>
+              Cancel
+            </Text>
+          )}
+          <Icon onClick={this.props.emojiPickerToggle} style={styleIcon} type="iconfont-emoji" />
```
Comment (buoyad) [inline shared/chat/conversation/input-area/normal/index.desktop.js]: 🍳 This doesn't seem to work

File: shared/chat/conversation/list-area/normal/container.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const hasExtraRow = !meta.resetParticipants.isEmpty() || !!meta.supersededByCausedBy
+  return {
+    conversationIDKey,
+    hasExtraRow,
+    messageOrdinals: Constants.getMessageOrdinals(state, conversationIDKey),
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _loadMoreMessages: (conversationIDKey: Types.ConversationIDKey) =>
```
Comment (buoyad) [inline shared/chat/conversation/list-area/normal/container.js]: available here too

File: shared/chat/conversation/list-area/normal/index.native.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    if (this.props.conversationIDKey !== nextProps.conversationIDKey) {
+      this.setState({calledLoadMoreDueToOrdinal: null})
+    }
+  }
+
+  render() {
+    return (
+      <ErrorBoundary>
+        <Box style={containerStyle}>
+          <NativeVirtualizedList
```
Comment (buoyad) [inline shared/chat/conversation/list-area/normal/index.native.js]: scroll is really nice on mobile again :+1:

File: shared/chat/conversation/info-panel/notifications/container.js (hunk 1 above)
Comment (chrisnojima) [reply shared/chat/conversation/info-panel/notifications/container.js]: not sure what you mean about this. don't you actually want this to be last?

File: shared/chat/conversation/input-area/mention-handler-hoc/index.desktop.js (hunk 2 above)
Comment (chrisnojima) [reply shared/chat/conversation/input-area/mention-handler-hoc/index.desktop.js]: fixed here in part 10: https://github.com/keybase/client/pull/10755/files#diff-e1e43dcc902e8130f21aadad81183422L148
due to the withState -> withStateHandlers update

File: shared/chat/conversation/input-area/channel-mention-hud/mention-hud-container.js (hunk 3 above)
Comment (chrisnojima) [reply shared/chat/conversation/input-area/channel-mention-hud/mention-hud-container.js]: fixed in 10: https://github.com/keybase/client/pull/10755/files#diff-70f2ef2567d7de29de82025ea06ec482R17

File: shared/chat/conversation/input-area/normal/index.desktop.js (hunk 5 above)
Comment (chrisnojima) [reply shared/chat/conversation/input-area/normal/index.desktop.js]: fixed in 10: https://github.com/keybase/client/pull/10755/files#diff-3697454f9700c7692f5234ab1832055dR301

File: shared/chat/conversation/input-area/mention-handler-hoc/index.native.js (hunk 4 above)
Comment (chrisnojima) [reply shared/chat/conversation/input-area/mention-handler-hoc/index.native.js]: fixed in 10: https://github.com/keybase/client/pull/10755/files#diff-17534f3bd85a2c3189f96e26e60b5bc1R51

**Comment by chrisnojima** [comment]
@buoyad updated, some in 10 ptal

**Pull request merged**